app/views/labware/_barcode_printing_form.html.erb

```diff
@@ -1,26 +1,50 @@
 <%= form_for(print_job, html: { method: :post, class: 'form-inline'}) do |f| %>
-  <%= f.hidden_field :label_template %>
+  <%= f.hidden_field :label_templates_by_service %>
 
   <% labels.each do |label| %>
+    <!-- main label for this plate -->
     <% label.attributes.each do |k, v|%>
       <%= f.hidden_field "labels[][main_label][#{k}]", value: v %>
     <%end%>
+    <!-- second label for 384-well plates -->
     <% label.extra_attributes.each do |k, v|%>
       <%= f.hidden_field "labels[][extra_label][#{k}]", value: v %>
     <%end%>
+    <!-- extra labels for intermediate plates in Duplex-Seq -->
     <% label.intermediate_attributes.each do |interm|%>
       <% interm.each do |k, v|%>
         <%= f.hidden_field "labels[][main_label][#{k}]", value: v %>
       <%end%>
     <%end%>
+    <!-- extra labels for QC plates in several pipelines -->
     <% label.qc_attributes.each do |qc_label|%>
       <% qc_label.each do |k, v|%>
         <%= f.hidden_field "labels[][main_label][#{k}]", value: v %>
       <%end%>
     <%end%>
   <% end %>
 
+  <% labels.each do |label| %>
+    <!-- this includes main label and extra label in one -->
+    <% label.sprint_attributes.each_with_index do |(k, v), index|%>
+      <%= f.hidden_field "labels_sprint[sprint][#{k}]", value: v %>
+    <%end%>
+    <!-- extra labels for intermediate plates in Duplex-Seq -->
+    <% label.intermediate_attributes.each_with_index do |interm, index|%>
+      <% interm.each do |k, v|%>
+        <%= f.hidden_field "labels_sprint[interm_#{index}][#{k}]", value: v %>
+      <%end%>
+    <%end%>
+    <!-- extra labels for QC plates in several pipelines -->
+    <% label.qc_attributes.each_with_index do |qc_label, index|%>
+      <% qc_label.each do |k, v|%>
+        <%= f.hidden_field "labels_sprint[qc_#{index}][#{k}]", value: v %>
+      <%end%>
+    <%end%>
+  <% end %>
+
   <%= f.label :printer_name, 'Barcode Printer', class: 'col-2' %>
+  <!-- pass the whole printer object through here, if poss, (instead of :name) or just the id and then re-query the printer through API -->
```

Comment:
Sorry, we left some of the comments a bit messy. The above one can go - will just be confusing as we've now implemented this